### PR TITLE
fix!: add forceRefresh parameter to FetchOrGetCachedWalletBalances endpoint

### PR DESCRIPTION
**Pull request by briansztamfater**
We have this issue on mobile (and possibly on desktop as well), marked as critical by our QA team:

https://github.com/status-im/status-mobile/issues/21725

It appears to be a balance cache issue. Currently, in the `fetchOrGetCachedWalletBalances` endpoint, we cannot force a refresh of balances to bypass the cache, nor is there a way to clear the cache from the client side. At the moment, the only way to clean the cache is by logging out and logging back in, which is not ideal.

This PR proposes adding a parameter to the endpoint that allows forcing a balance refresh. The parameter will be passed to the reader, where the forceRefresh functionality is already implemented.

**Note:** This is a breaking change, so both mobile and desktop clients will need to adapt to it if this PR is merged.

### CODE REVIEW

**Review by saledjenic — APPROVED** [review]

**Review by shivekkhurana — APPROVED** [review]

**Review by clauxx — APPROVED** [review]

**Review by vkjr — APPROVED** [review]

**Comment by dlipicar** [comment]
This is a breaking change, so the commit message should have an "!" , not an "_"

File: protocol/messenger_wallet.go
Comment (smohamedjavid) [inline protocol/messenger_wallet.go:53]: Not for this PR. Is this something we need to cleanup as don't actually use the response of the method and we don't signal the tokens?
cc @saledjenic

**Review by smohamedjavid — APPROVED** [review]

File: protocol/messenger_wallet.go
Comment (saledjenic) [reply protocol/messenger_wallet.go:53]: I am not that much into this code, defer this to @alaibe @Khushboo-dev-cpp @dlipicar

File: services/wallet/api.go
Comment (ilmotta) [inline services/wallet/api.go:87]: This PR is a good example of why it's important to use a request map instead of position args because then it would be possible to add a new struct field for `forceRefresh` and not break any client.

IMO it would be great if we followed this standard by the book because when endpoints are designed initially we cannot anticipate which other args it may need. I remember a handful of cases where such breaking changes were introduced in status-go PRs and could've been avoided. cc @igor-sirotin

Comment (ilmotta) [reply services/wallet/api.go:87]: @briansztamfater I'm not suggesting changes for this PR because now it doesn't matter, too late because either way we do we break. But in the future we can totally avoid this type of breakage.

Comment (briansztamfater) [reply services/wallet/api.go:87]: Totally agree @ilmotta 💯 , I didn't want to refactor the whole implementation given the timing for the upcoming release (and my little expertise with Go), but a map would have made this change much simpler to handle for the clients and not breaking.

**Comment by briansztamfater** [comment]
Updated the PR with unit test, summoning some Go experts to re-review before merging cc @saledjenic @dlipicar @friofry

File: protocol/messenger_wallet.go
Comment (briansztamfater) [reply protocol/messenger_wallet.go:53]: Resolving this conversation to pass CI checks

**Comment by saledjenic** [comment]
> Updated the PR with unit test, summoning some Go experts to re-review before merging cc @saledjenic @dlipicar @friofry

@briansztamfater you already have approvals. If mobile QAs approve it, feel free to merge it (the desktop is ready to adapt to the change).

**Pull request merged**
